packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -713,38 +713,55 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
         });
     }
 
-    private findTabBar(event?: Event): TabBar<Widget> | undefined {
-        if (event && event.target) {
-            const tabBar = this.shell.findWidgetForElement(event.target as HTMLElement);
-            if (tabBar instanceof TabBar) {
-                return tabBar;
-            }
-        }
-        return this.shell.currentTabBar;
-    }
-
     private findTabArea(event?: Event): ApplicationShell.Area | undefined {
-        const tabBar = this.findTabBar(event);
+        const tabBar = this.shell.findTabBar(event);
         if (tabBar) {
             return this.shell.getAreaFor(tabBar);
         }
         return this.shell.currentTabArea;
     }
 
-    private findTitle(tabBar: TabBar<Widget>, event?: Event): Title<Widget> | undefined {
+    /**
+     * Evaluates the currentIndex of the title in the array of titles.
+     * @param event: `event` to be used when searching for the title and the tab-bar.
+     *
+     * @returns `currentIndex` if the `targetTitle` is available in the array, else returns the index of currently-selected title.
+     */
+    private targetTitleIndex(event?: Event): number {
+        const tabBar = this.shell.findTabBar(event)!;
+        const targetTitle = this.shell.findTitle(tabBar, event);
+        let currentIndex: number;
+        if (targetTitle) {
+            currentIndex = tabBar.titles.indexOf(targetTitle);
+        } else {
+            currentIndex = tabBar.currentIndex;
+        }
+        return currentIndex;
+    }
+
+    private canToggleMaximized(event?: Event): boolean {
         if (event && event.target) {
-            let tabNode: HTMLElement | null = event.target as HTMLElement;
-            while (tabNode && !tabNode.classList.contains('p-TabBar-tab')) {
-                tabNode = tabNode.parentElement;
+            const widget = this.shell.findWidgetForElement(event.target as HTMLElement);
+            if (widget) {
+                return this.shell.mainPanel.contains(widget) || this.shell.bottomPanel.contains(widget);
             }
-            if (tabNode && tabNode.title) {
-                const title = tabBar.titles.find(t => t.label === tabNode!.title);
-                if (title) {
-                    return title;
+        }
+        return false;
+    }
+
+    private toggleMaximized(event?: Event): void {
```

Comment:
Should 'Toggle Maximized' reveal the editor/view after it is is maximized?
I find the UX odd that we trigger a context-menu for a given tab, select 'Toggle Maximized' for the given tab and it is not selected for me. It will always require two clicks for an tab that is not previously selected.

Reply:
Yes, you're right, we should toggle the tab and maximize the specific tab in this case, right?